File: samples/sample.py
import os
import shutil

class FileExplorer:
    def __init__(self, path="."):
        self.path = path

    def list_files(self):
        print("Files in directory:", self.path)
        for filename in os.listdir(self.path):
            print(filename)

    def list_directories(self):
        print("Directories in directory:", self.path)
        for dirname in os.listdir(self.path):
            if os.path.isdir(dirname):
                print(dirname)

    def create_directory(self, dirname):
        try:
            os.mkdir(os.path.join(self.path, dirname))
            print("Directory created successfully.")
        except OSError as e:
            print("Directory creation failed:", e)

    def delete_directory(self, dirname):
        try:
            shutil.rmtree(os.path.join(self.path, dirname))
            print("Directory deleted successfully.")
        except OSError as e:
            print("Directory deletion failed:", e)

    def copy_file(self, src_file, dest_file):
        try:
            shutil.copy(src_file, dest_file)
            print("File copied successfully.")
        except OSError as e:
            print("File copy failed:", e)

    def move_file(self, src_file, dest_file):
        try:
            shutil.move(src_file, dest_file)
            print("File moved successfully.")
        except OSError as e:
            print("File move failed:", e)

if __name__ == '__main__':
    explorer = FileExplorer()
    explorer.list_files()
    explorer.list_directories()
    explorer.create_directory("new_directory")
    explorer.delete_directory("new_directory")
    explorer.copy_file("test.txt", "test_copy.txt")
    explorer.move_file("test_copy.txt", "new_directory/test_copy.txt")

File: samples/sample.js
const fs = require('fs');
const path = require('path');

class FileManager {
  constructor(currentDirectory) {
    this.currentDirectory = currentDirectory || '.';
  }

  list() {
    const files = fs.readdirSync(this.currentDirectory);
    console.log('Files in current directory:');
    files.forEach(file => {
      console.log(file);
    });
  }

  createDirectory(directoryName) {
    fs.mkdirSync(path.join(this.currentDirectory, directoryName));
    console.log(`${directoryName} created successfully!`);
  }

  deleteDirectory(directoryName) {
    fs.rmdirSync(path.join(this.currentDirectory, directoryName), { recursive: true });
    console.log(`${directoryName} deleted successfully!`);
  }

  createFile(fileName, content) {
    fs.writeFileSync(path.join(this.currentDirectory, fileName), content);
    console.log(`${fileName} created successfully!`);
  }

  deleteFile(fileName) {
    fs.unlinkSync(path.join(this.currentDirectory, fileName));
    console.log(`${fileName} deleted successfully!`);
  }

  copyFile(sourceFileName, destinationFileName) {
    fs.copyFileSync(path.join(this.currentDirectory, sourceFileName), path.join(this.currentDirectory, destinationFileName));
    console.log(`${sourceFileName} copied to ${destinationFileName} successfully!`);
  }

  moveFile(sourceFileName, destinationFileName) {
    fs.renameSync(path.join(this.currentDirectory, sourceFileName), path.join(this.currentDirectory, destinationFileName));
    console.log(`${sourceFileName} moved to ${destinationFileName} successfully!`);
  }
}

// Example Usage
const fileManager = new FileManager();
fileManager.list();
fileManager.createDirectory('newDirectory');
fileManager.createFile('newFile.txt', 'This is a new file!');
fileManager.copyFile('newFile.txt', 'newFileCopy.txt');
fileManager.moveFile('newFile.txt', 'newDirectory/newFile.txt');
fileManager.deleteFile('newFile.txt');
fileManager.deleteDirectory('newDirectory');
